src/dowel/csv_output.py

```diff
@@ -41,13 +42,25 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
-            if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+            # add new keys to _fieldnames set
+            old_header_size = len(self._fieldnames)
+            for key in to_csv.keys():
+                self._fieldnames.add(key)
+
+            # if new keys are added, rewrite the csv file
+            if len(self._fieldnames) > old_header_size:
+                self._writer = csv.DictWriter(
+                    self._log_file,
+                    fieldnames=sorted(list(self._fieldnames)),
+                    extrasaction='ignore')
```

Comment:
Why use `extrasaction='ignore'` here? It seems like `extrasaction='raise'` would be a safer choice.

Reply:
Agree, fixed in new commit. Although I suspect because of the header update logic above, it is unlikely that self._writer will see an extra key now. It is still safer to change to 'raise', so we know if this breaks in future.